### PR TITLE
FIX: AMS ID is on the TopLevelProject

### PR DESCRIPTION
**Pull request by klauer**
cc @zllentz - as we ran into this yesterday.

Output from the project during the demo:
```
$ pytmc summary --plcs tests/projects/motion-demo/motion-demo/motion-demo/motion-demo.tsproj
PLC Project (1): Demo1
======================

    Project path: /Users/klauer/docs/Repos/pytmc/tests/projects/motion-demo/motion-demo/motion-demo/Demo1/Demo1.plcproj
    TMC path:     /Users/klauer/docs/Repos/pytmc/tests/projects/motion-demo/motion-demo/motion-demo/Demo1/Demo1.tmc
    AMS ID:       5.57.239.146.1.1
    IP Address:   5.57.239.146 (* based on AMS ID)
    Port:         851

    Source files:
        1.) /Users/klauer/docs/Repos/pytmc/tests/projects/motion-demo/motion-demo/motion-demo/Demo1/PlcTask.TcTTO
        2.) /Users/klauer/docs/Repos/pytmc/tests/projects/motion-demo/motion-demo/motion-demo/Demo1/POUs/Main.TcPOU

    POUs:
        1.) Main

PLC Project (2): Demo2
======================

    Project path: /Users/klauer/docs/Repos/pytmc/tests/projects/motion-demo/motion-demo/motion-demo/Demo2/Demo2.plcproj
    TMC path:     /Users/klauer/docs/Repos/pytmc/tests/projects/motion-demo/motion-demo/motion-demo/Demo2/Demo2.tmc
    AMS ID:       5.57.239.146.1.1
    IP Address:   5.57.239.146 (* based on AMS ID)
    Port:         852

    Source files:
        1.) /Users/klauer/docs/Repos/pytmc/tests/projects/motion-demo/motion-demo/motion-demo/Demo2/PlcTask.TcTTO
        2.) /Users/klauer/docs/Repos/pytmc/tests/projects/motion-demo/motion-demo/motion-demo/Demo2/POUs/Main.TcPOU

    POUs:
        1.) Main
```

### CODE REVIEW

**Comment by ZLLentz** [comment]
I'm still surprised (but also not really surprised) that these projects only sometimes have the ams net id on the lower files... Hopefully 100% of them have it at the top tsproj location.

**Comment by klauer** [comment]
I reviewed the `.tsproj` files I have here and it _appears_ to be consistent. It may have been due to refactoring what `Project` means that I just messed it up.

**Comment by ZLLentz** [comment]
That's fair, TwinCAT definitely reuses the keyword "project" way too many times

File: pytmc/parser.py [hunk 1]
```diff
@@ -324,7 +324,24 @@ def __repr__(self):
...
 class TopLevelProject(TwincatItem):
-    '[tsproj] '
+    '[tsproj] Containing Io, System, Motion, TopLevelPlc, etc.'
+
+    @property
+    def ams_id(self):
+        '''
+        The AMS ID of the configured target
+        '''
+        return self.attributes.get('TargetNetId', '')
```
Comment (teddyrendahl) [inline pytmc/parser.py]: Missing the `AMS` ID seems important enough we might want to throw an error out here?

Comment (klauer) [reply pytmc/parser.py]: I'd rather the objects themselves (TopLevelProject) not be raising exceptions in such cases, I think. Perhaps just in st.cmd generation we verify the target net ID exists, and (optionally?) blow up if unavailable. Seem reasonable?

Comment (teddyrendahl) [reply pytmc/parser.py]: Yeah a`property` that raises an error always seem slightly distasteful. Find with checking in the flow of the `st.cmd` generation, but I think I want a loud failure at some point there.

Comment (klauer) [reply pytmc/parser.py]: OK, I think I have addressed that. It's possible we could also allow specification of IP/AMS ID from the tool itself down the line 🤷‍♂

Comment (teddyrendahl) [reply pytmc/parser.py]: Right. You may not have written and built the code on the PLC you plan to deploy?

**Comment by ZLLentz** [comment]
FYI this works 100% with my latest project

File: pytmc/parser.py (hunk 1 above)
Comment (klauer) [reply pytmc/parser.py]: Mind making an issue for this feature?

**Comment by klauer** [comment]
Force-pushed to rebase and fix style error. I think this is ready to merge once tests pass, and should be done prior to tagging.

**Pull request merged**